src/main/resources/de/itemis/mps/gradle/mpssign.sh

```diff
@@ -71,13 +86,20 @@ if [[ -d "$HELP_DIR" ]]; then
     hiutil -Cagvf "$HELP_DIR/search.helpindex" "$HELP_DIR"
 fi
 
-# Make sure JetBrainsMacApplication.p12 is imported into local KeyChain
-#security unlock-keychain -p <password> /Users/builduser/Library/Keychains/login.keychain
-#codesign -v --deep -s "Developer ID Application: JetBrains" "$OUTPUT_DIR/$BUILD_NAME"
-#echo "signing is done"
-#echo "check sign"
-#codesign -v "$OUTPUT_DIR/$BUILD_NAME" -vvvvv
-#echo "check sign done"
+# Make sure your certificate is imported into local KeyChain
+if [[ -n "$SIGN_PW" && -n "$SIGN_KEY_CHAIN" && -n "$SIGN_IDENTITY" ]]; then
+    echo "Signing application $BUILD_NAME"
+    echo "key chain: $SIGN_KEY_CHAIN"
+    echo "sign identity: $SIGN_IDENTITY"
+    security unlock-keychain -p $SIGN_PW $SIGN_KEY_CHAIN
+    codesign -v --deep -s "$SIGN_IDENTITY" "$OUTPUT_DIR/$BUILD_NAME"
+    echo "signing is done"
+    echo "check sign"
+    codesign -v "$OUTPUT_DIR/$BUILD_NAME" -vvvvv
+    echo "check sign done"
+else
+    echo "for signing the application $BUILD_NAME: SIGN_PW, SIGN_KEY_CHAIN and SIGN_IDENTITY needs to be provided"
```

Comment:
Do we need all three here? I think it's enough when just the `SIGN_IDENTITY` is present. From the `codesign`docs it will search by default in all keychains the user has access to:

> During signing, only search for the signing identity in the keychain file specified. This can be used to break any matching ties if you have multiple simi-
             larly-named identities in several keychains on the user's search list.  Note that the standard keychain search path is still consulted while constructing the
             certificate chain being embedded in the signature.
             Note that filename will not be searched to resolve the signing identity's certificate chain unless it is also on the user's keychain search list.

The `SIGN_KEY_CHAIN` argument would only be required incase where a) the keychain is not unlocked automatically in which case also require `SIGN_PW` to unlock it  or b) where a identity matching the `SIGN_KEY_CHAIN` is present in more than one keychain. 

AFAIK the default setting of mac OS that the keychain is unlocked on login for read access.

From understanding that would only make `SIGN_IDENTITY` a required parameter.  `SIGN_KEY_CHAIN` and `SIGN_PW` are optional. 

If `SIGN_KEY_CHAIN` is present and no `SIGN_PW` is present then we do not unlock the keychain. 
If `SIGN_PW` is present `SIGN_KEY_CHAIN` is also required and we unlock the keychain. 

Do we need to handle this a both places or is it enough if we only do in the gradle front end? Seems a bit odd tu duplicate the error handling logic inside the bash script and the gradle frontend for the user.

Reply:
I would keep the error handling at both places since the bash script is sort of self-contained. Don't know about the required vs optional parameters, let's just keep this as is since it's working for Alex.